front/src/common/api/generatedEditoastApi.ts

```diff
@@ -3443,6 +3443,22 @@ export type PathfindingResultSuccess = {
   /** Path description as track ranges */
   track_section_ranges: TrackRange[];
 };
+export type OffsetRange = {
+  end: number;
+  start: number;
+};
+export type IncompatibleOffsetRangeWithValue = {
+  range: OffsetRange;
+  value: string;
```

Comment:
This could have been like this @bougue-pe ?

```suggestion
export type IncompatibleOffsetRangeWithValue = IncompatibleOffsetRange & {
  value: string;
};
```

Reply:
After discussion, it probably means using `#[serde(flatten)]` on the range in rust, and maybe a change in core too.
I'll try to check it (today, but the first one trying just drops a word to avoid multiple work).

Reply:
I had a look to see if it's easily/cleanly doable: yes for editoast (`#[serde(flatten)]`), no for core 🫤
In core we would need to either:
* re-declare the range part without sharing
* tweak the moshi serializer
* abandon the data class that can't be inherited
* ... didn't think of any other possibility?
All in all, it bloats core way more than the gain in editoast and front, so I'd let it like that

Reply:
Ok I understand

Reply:
For most of the cases I checked, yes (no direct serializing of something provided by core).
The only counter-example I found is `CompleteReportTrain` (`ReportTranV2`) which uses inheritance in core, and wouldn't take advantage of a data class.